streaming/multimodal/webvid.py

```diff
@@ -40,3 +44,195 @@ def __getitem__(self, idx: int) -> Any:
+            iterating. Defaults to ``100_000``.
+        keep_zip (bool, optional): Whether to keep or delete the compressed file when
+            decompressing downloaded shards. If set to None, keep iff remote is local. Defaults to
+            ``None``.
+        download_retry (int): Number of download re-attempts before giving up. Defaults to ``2``.
+        download_timeout (float): Number of seconds to wait for a shard to download before raising
+            an exception. Defaults to ``60``.
+        validate_hash (str, optional): Optional hash or checksum algorithm to use to validate
+            shards. Defaults to ``None``.
+        shuffle_seed (int): Seed for Deterministic data shuffling. Defaults to ``9176``.
+        num_canonical_nodes (int, optional): Canonical number of nodes for shuffling with resumption.
+            Defaults to ``None``, which is interpreted as the number of nodes of the initial run.
+        batch_size (int, optional): Batch size of its DataLoader, which affects how the dataset is
+            partitioned over the workers. Defaults to ``None``.
+        extra_local (str, optional): Base destination of extra local sample downloads.
+        extra_remote (str, optional): Base source of extra remote sample downloads.
+    """
+
+    def __init__(self,
+                 local: str,
+                 remote: Optional[str] = None,
+                 split: Optional[str] = None,
+                 shuffle: bool = False,
+                 predownload: Optional[int] = 100_000,
+                 keep_zip: Optional[bool] = None,
+                 download_retry: int = 2,
+                 download_timeout: float = 60,
+                 validate_hash: Optional[str] = None,
+                 shuffle_seed: int = 9176,
+                 num_canonical_nodes: Optional[int] = None,
+                 batch_size: Optional[int] = None,
+                 extra_local: Optional[str] = None,
+                 extra_remote: Optional[str] = None):
+        super().__init__(local, remote, split, shuffle, predownload, keep_zip, download_retry,
+                         download_timeout, validate_hash, shuffle_seed, num_canonical_nodes,
+                         batch_size)
+        self.extra_local = extra_local
+        self.extra_remote = extra_remote
+
+    def __getitem__(self, idx: int) -> Any:
+        obj = super().__getitem__(idx)
+
+        if self.extra_local and self.extra_remote:
+            rel_path = obj['content_path']
+            local = os.path.join(self.extra_local, rel_path)
+            remote = os.path.join(self.extra_remote, rel_path)
+            if not os.path.exists(local):
+                download(remote, local, self.download_timeout)
+            with open(local, 'rb') as fp:
+                content = fp.read()
+            obj['content'] = content
+
+        # Processing goes here.
+
+        return obj
+
+
+class StreamingOutsideDTWebVid(StreamingDataset):
+    """Streaming WebVid dataset.
```

Comment:
Please add more detail description. See my above comment for `StreamingOutsideGIWebVid`.